src/sql/magic.py

```diff
@@ -640,6 +674,38 @@ def _persist_dataframe(
         )
 
 
+def get_query_type(command: str):
+    """
+    Returns the query type of the original sql command
+    """
+    query_type = (
+        sqlparse.parse(command)[0].get_type() if sqlparse.parse(command) else None
+    )
+    if query_type == "UNKNOWN":
+        return None
+    return query_type
+
+
+def get_dependency_in_CTE(sql_original, dependencies):
+    """
+    The following bit of code checks for dependencies within the parentheses
+    We do not care about the name outside the parentheses being the same as
+    another table / snippet here and only check if there are dependencies
+    in the parentheses.
+    This is for generating more relevant warnings.
+    """
+
```

Comment:
I'm confused about what this function does. based on the code, this is executed when the `is_cte` flag is True, hence, it executes when the user's query begins with `WITH`, and it'll parse anything that's within parenthesis

e.g.,

```sql
with something (
 /* grab everything from here */
)
```

is this right?

if so, I'm confused because of the unit tests has a `CREATE TABLE AS ( ...)` as input, and it doesn't seem like this function will ever be executed with a `CREATE TABLE AS (...)` statement

Also, the unit tests only contain check when the CTE contains a single statement, but this implementation will yield weird results when there is more than one:


```
In [8]: sql_original = """
   ...:
   ...: with something as(
   ...: a
   ...: ), another as(
   ...: b)
   ...: """

In [9]:

In [9]: parentheses_content = re.findall(r"\((.*)\)", sql_original.replace("\n", " "))

In [10]: parentheses_content
Out[10]: [' a ), another as( b']
```

do we want `parenthesis_content` to be equal to `[' a ), another as( b']`? seems like an odd result.

I think at this point, it might be better to just delete this code and show a generic warning, because from what I see, the only purpose of this is to show better warnings.

Reply:
@edublancas I did write this logic for better warning only. I can improve it if you think we need detailed warnings, otherwise I can show a generic one.

Reply:
@edublancas I've removed the logic and provided a more generic warning instead

Reply:
[Fixed](https://github.com/ploomber/jupysql/pull/878/commits/a204e262aa93b6bf06494dd2e06c1f27b25e0878)